cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// legacy_boot allows to handover a system running linuxboot/u-root
+// to a legacy preinstalled operating system by replacing the traditional
+// bootloader path
+
+//
+// Synopsis:
+//	legacy_boot
+//
+// Description:
+//	If returns to u-root shell, the code didn't found a local bootable option
+//
+// Notes:
+//	The code is looking for boot/grub/grub.cfg file as to identify the
+//	boot option.
+//	The first bootable device found in the block device tree is the one used
+//	Windows is not supported (that is a work in progress)
+//
+// Example:
+//	legacy_boot -v 	- Start the script in verbose mode for debugging purpose
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/u-root/u-root/pkg/kexec"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"flag"
+	"syscall"
+)
+
+var verbose bool
+type options struct {
+        verbose bool
+}
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func blk_devices_list(blkpath string, devpath string) []string {
+	var blk_devices []string
+	files, err := ioutil.ReadDir(blkpath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		device_entry, err := ioutil.ReadDir(blkpath + file.Name() + devpath)
+		if err == nil {
+			blk_devices = append(blk_devices, device_entry[0].Name())
+		}
+	}
+	return (blk_devices)
+}
+
+// Current support is limited to Hard disk devices and USB devices
+
+func check_for_bootable_mbr(path string) int {
+	var b511, b510 byte
+	f, err := os.Open(path)
+	check(err)
+	b1 := make([]byte, 512)
+	f.Read(b1)
+	b511 = b1[511]
+	b510 = b1[510]
+	f.Close()
+	if ((b511 == 0xaa) && (b510 == 0x55)) == true {
```

Comment:
what are these two magic values? would prefer some constants, both for the 510 and 511 (field names?) and the values.